src/map/battle.cpp

```diff
@@ -3353,25 +3356,67 @@ static bool attack_ignores_def(struct Damage* wd, struct block_list *src, struct
 	return nk[NK_IGNOREDEFENSE] != 0;
 }
 
-/*================================================
- * Should skill attack consider VVS and masteries?
- *------------------------------------------------
- * Credits:
- *	Original coder Skotlex
- *	Initial refactoring by Baalberith
- *	Refined and optimized by helvetica
+/**
+ * This function lists which skills are unaffected by refine bonus, masteries, Star Crumbs and Spirit Spheres
+ * This function is also used to determine if atkpercent applies
+ * @param skill_id: Skill being used
+ * @param type 1 - Checking refine bonus; 2 - Checking Star Crumb bonus
+ * @return true = bonus applies; false = bonus does not apply
  */
-static bool battle_skill_stacks_masteries_vvs(uint16 skill_id)
+static bool battle_skill_stacks_masteries_vvs(uint16 skill_id, int type)
```

Comment:
May be just me, but this whole function sounds like a flag that should be put into skill_db.yml

Reply:
Eventually yes, but I won't mess with YML myself.

Reply:
Okay, please open an issue for these two after merging then, so that someone can take it up.

Reply:
Could already do this now (then I can mark the comment as resolved).

But it's a bit more complex because there are a few special skills with hard-coded custom behaviors that go against the norm such as Grand Cross (considers refine but nothing else) and Spiral Pierce (considers refine and Star Crumbs, but nothing else).

Also some of the existing flags probably also require some rework. I think the "IgnoreElement" flag for example is kinda weird, because it makes you ignore element cards but not other cards. But the one weapon skill that even uses this flag "Venom Splasher" actually ignores all card modifiers but shouldn't even ignore the attribute table (I created an issue on that one already).

It would be better to have more test data on all skills to see the full picture first. But I guess I can just set the "Needs more info" tag on the issue and explain it in the issue.

Reply:
Feature request added: https://github.com/rathena/rathena/issues/8226